core/tx_pool.go

```diff
@@ -1459,25 +1582,33 @@ func (pool *TxPool) truncatePending() {
 
 	// If still above threshold, reduce to limit or min allowance
 	if pending > pool.config.GlobalSlots && len(offenders) > 0 {
+		var (
+			list *txList
+			hash common.Hash
+		)
+
 		for pending > pool.config.GlobalSlots && uint64(pool.pending[offenders[len(offenders)-1]].Len()) > pool.config.AccountSlots {
 			for _, addr := range offenders {
-				list := pool.pending[addr]
+				list = pool.pending[addr]
 
-				caps := list.Cap(list.Len() - 1)
+				caps = list.Cap(len(list.txs.items) - 1)
```

Comment:
Please check is it equal to Len(caps) that you use later

Reply:
done